_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,165 @@
+
+On each **TeamCity** build agent we should have installed **Docker**. We will also use this machine to run our application. In a real project we don't want to install **TeamCity** agents on all machines and we should introduce another tool (e.g. Kubernetes) which will take care about distribution of our application.
+ 
+## Build
+
+We need to build something before we will be able to create any docker image. To do that we need to create the first step in our Continuous Deployment (CD) pipeline. 
+In this step we will download our source code, run all **tests** which can be done on the source code level and produce an **artifact** containing all elements required to start and run our application.
+
+This build configuration is not very different to a usual step in a common Continuous Deployment pipeline. Alongside common parameters we have to define **artifacts** which will be generated after each execution. We are going to use them as a base for next steps in the pipeline.
+
+In **TeamCity** we define **artifacts** by defining paths to files from the working directory (which is created for each run of a configuration). The working directory is combination of files downloaded from a version control system and files generated during the execution of build steps. We will define these elements during configuring a version control settings and build steps. 
+
+We need two artifacts. The first one is `Dockerfile`. We already prepared this file in our source code and it will be stored in `docker` directory.
+The next file is a ***tar*** file which will be generated by our **Gradle** build. It contains a script which allows to execute our code and all required libraries. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_config.png" alt="Build Configuration" title="Build Configuration" class="img img-responsive style-screengrab">
+
+Now we are ready to instruct our build configuration how to download our source code. We use GitHub as our code repository.
+We just have to choose type as *Git* and provide the URL to our application.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_vcs.png" alt="Version Control Configuration" title="Version Control Configuration" class="img img-responsive style-screengrab">
+
+We have the source code, so we can describe what to do with this code. As I mentioned before we will **test** our application and generate an **artifact** with executable content.   
```

Comment:
It is not executable artifact. It contains executable script and libraries.

`As I mentioned before we run all the **tests** and generate an **artifact** containing the start script and libraries.`